client/lobbies/lobby-reducer.js

```diff
@@ -153,6 +154,12 @@ const infoReducer = keyedReducer(undefined, {
   [NETWORK_SITE_CONNECTED](state, action) {
     return new LobbyInfo()
   },
+
+  [MAPS_TOGGLE_FAVORITE](state, action) {
```

Comment:
Well, right now the lobby state contains the whole MapRecord of the chosen map. I guess we could update it to only contain a map ID or something, and keep the actual map info in the maps reducer, but I feel like this would complicate things a lot. Currently, the maps reducer is only used by the server-maps-browser (I actually thought of renaming that reducer to server-maps-reducer, since we also have a local-maps-reducer), and every other place that fetches maps keep them in their own reducers, e.g. find-match, preferred-maps, matchmaking-map-pools, create-lobby, lobby, etc.

When I was working on this, I did try to think of ways to keep the maps state in one place, but it proved too hard for me. Maybe I was missing something though :/

Reply:
It definitely feels to me like this data *should* be in one place, indexable by an ID or whatever. Otherwise every place that involves a map has to have its own way of getting that data from the server, its own way of updating that data (favoriting, editing, etc), has to handle all of the various map events to ensure the data stays up-to-date, etc.

I think the idea of handling a single event in many different ways throughout our reducers is sound, but when that handling boils down to the exact same desired outcome on nearly the exact same data structure, it's far better to put all that logic/data in a single spot.

I'm fine with this being this way for this PR, since it already exists this way, but it's something we should be looking to move away from imo